src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -177,18 +211,56 @@ public Map<String, Object> getProcessedRecordForStreamingIngest(SinkRecord recor
     final Map<String, Object> streamingIngestRow = new HashMap<>();
     for (JsonNode node : row.content.getData()) {
       try {
-        streamingIngestRow.put(TABLE_COLUMN_CONTENT, MAPPER.writeValueAsString(node));
+        if (enableSchematization) {
```

Comment:
See my earlier comment for extracting this into a separate method in record service. Lot of things going on this method. IMO, deserves a new method.